src/renderer/components/item-object-list/item-list-layout.tsx

```diff
@@ -111,28 +112,49 @@ export class ItemListLayout extends React.Component<ItemListLayoutProps> {
   }
 
   async componentDidMount() {
+    disposeOnUnmount(this, [
+      namespaceStore.onContextChange(() => this.loadStores(), {
```

Comment:
`ItemListLayout` definitely needs to be refactored, but not part of this PR.

Reply:
But the changes you're adding in this PR, can they be done in `kube-object-list-layout`?

Reply:
Mostly no.. since `namespace` already introduced in `ItemsListLayout` and what's done in PR is just a small refactoring for loading stores + added "event-handler" for updating data related again to namespaces:

```ts
    disposeOnUnmount(this, [
      namespaceStore.onContextChange(() => this.loadStores(), {
        fireImmediately: true,
      })
    ]);
```